### PR TITLE
curl: writeout: fix repeated header outputs

### LINKED ISSUE #9150: `-w %header{name}` prints broken string on the second request
### I did this

```
$ curl https://github.com/curl/curl -s -o /dev/null -w "%header{date}\n"
Wed, 13 Jul 2022 07:03:27 GMT
```
but,
```
$ curl https://github.com/curl/curl#[1-2] -s -o /dev/null -w "%header{date}\n"
Wed, 13 Jul 2022 07:03:44 GMT
%header{date
```

### I expected the following

```
$ curl https://github.com/curl/curl#[1-2] -s -o /dev/null -w "%header{date}\n"
Wed, 13 Jul 2022 07:03:44 GMT
Wed, 13 Jul 2022 07:03:44 GMT
```

### curl/libcurl version

curl 7.84.0 (x86_64-pc-cygwin) libcurl/7.84.0 OpenSSL/1.1.1q zlib/1.2.12 brotli/1.0.9 zstd/1.5.2 libidn2/2.3.2 libpsl/0.21.1 (+libidn2/2.3.2) libssh2/1.10.0 nghttp2/1.48.0
Release-Date: 2022-06-27
Protocols: dict file ftp ftps gopher gophers http https imap imaps mqtt pop3 pop3s rtsp scp sftp smb smbs smtp smtps telnet tftp
Features: alt-svc AsynchDNS brotli GSS-API HSTS HTTP2 HTTPS-proxy IDN IPv6 Kerberos Largefile libz NTLM NTLM_WB PSL SPNEGO SSL threadsafe TLS-SRP UnixSockets zstd

### operating system

MSYS_NT-10.0-19044 hostname 3.3.5-341.x86_64 2022-07-04 21:33 UTC x86_64 Msys
on
Windows 10 Pro 21H2 19044.1766

### PR DESCRIPTION
**Pull request by bagder**
The function stores a terminating zero in the buffer for convenience, but
when on repeated calls that would cause problems. Starting now, the original
content is put back again and repeated calls work fine.

Fixes #9150
Reported-by: highmtworks on github

### CODE REVIEW

**Comment by jay** [comment]
Though we already do this save and restore elsewhere and it works fine I really don't like it, it is not correct because ptr is derived from `const char *` argument and we are stripping the const. It just happens to work the way we use it now since we aren't passing const. I think the easiest way to remedy this would be update the prototype and I've added a commit for that.

**Comment by bagder** [comment]
I don't think we should remove the const but rather consider making the code not write to it.

**Comment by bagder** [comment]
I now made the function not touch the passed in buffer.

File: src/tool_writeout.c [hunk 1]
```diff
@@ -380,21 +380,26 @@ void ourWriteOut(const char *writeinfo, struct per_transfer *per,
             }
           }
           if(!match) {
-            fprintf(stderr, "curl: unknown --write-out variable: '%s'\n", ptr);
+            fprintf(stderr, "curl: unknown --write-out variable: '%.*s'\n",
+                    (int)vlen, ptr);
           }
           ptr = end + 1; /* pass the end */
```
Comment (jay) [inline src/tool_writeout.c:386]: typo here as well, suggest remove comment, otherwise lgtm

**Review by jay — APPROVED** [review]

File: src/tool_writeout.c (hunk 1 above)
Comment (bagder) [reply src/tool_writeout.c:386]: But the comment is unrelated to this change, I prefer not to modify it (now)

**Pull request closed without merging**